client/src/featureform/wait_test.py

```diff
@@ -0,0 +1,158 @@
+import pytest
+
+from datetime import timedelta
+
+
+TRAININGSET_NAME = "test_training_set"
+TRAININGSET_VARIANT = "test_ts_variant"
+
+FEATURE_NAME = "test_feature_name"
+FEATURE_VARIANT = "test_feature_variant"
+
+
+
+#Training set
+
+# not ready, expect timeout
+def test_not_ready_timeout_training_set(mocker):
+    mocker.patch(
+        # api_call is from slow.py but imported to main.py
+        'featureform.register.get_training_set',
+        return_value=TrainingSet(
+            name="",
+            owner="",
+            label=("",""),
+            features=[("","")],
+            feature_lags=[],
+            description="",
+            status="PENDING"
+        )
+    )
+
+    try:
+        training_set = dataset = client.training_set(TRAININGSET_NAME, TRAININGSET_VARIANT).wait(timeout=timedelta(seconds=1))
```

Comment:
We might want to just make this an int instead of timedelta. Looks pretty ugly like this